ocaml/forkexecd/test/fe_test.ml

```diff
@@ -109,14 +109,18 @@ let test_delay () =
   let start = Unix.gettimeofday () in
   let exe = Printf.sprintf "/proc/%d/exe" (Unix.getpid ()) in
   let args = ["sleep"] in
+  let timeout = 3.7 in
```

Comment:
Any reason to change it to this particular value? try `1.3`, should test for values different that full seconds and be faster

Reply:
An alternate approach would be to use `mtime` and `mtime.clock.os` libraries, which I've added to my mtime branch:

```ocaml
let is_shorter s ~than = compare s than < 0

let is_longer s ~than = compare s than > 0

type in_range = In_range | Longer | Shorter

let in_range ~e ~around span =
  let lower = Mtime.Span.abs_diff around e in
  let upper = Mtime.Span.add around e in
  if is_shorter ~than:lower span then
    Shorter
  else if is_longer ~than:upper span then
    Longer
  else
    In_range
```
```ocaml
let test_delay () =
  let start = Mtime_clock.counter () in
  let timeout = Mtime.Span.(1300 * ms) in
```
```ocaml
  | Forkhelpers.Subprocess_timeout -> (
      let elapsed = Mtime_clock.count start in
      Printf.printf "Caught timeout exception after %s\n%!"
        (Fmt.to_to_string Mtime.Span.pp elapsed) ;

      match in_range ~e:Mtime.Span.(200 * ms) ~around:timeout elapsed with
      | In_range ->
          ()
      | Shorter ->
          failwith "Process exited too soon"
      | Longer ->
          failwith "Process took too long to exit"
    )
```

Reply:
> Any reason to change it to this particular value? try `1.3`, should test for values different that full seconds and be faster

Yes, there's a reason. While I was doing my changes/experiment an internal computation related to sub-seconds (fractional part) and overflow on fractional part (POSIX APIs separate seconds and fractional parts). Surely worth adding a comment. So the initial `4.0` timeout was not good to detect this. For the fractional part better to use a value high (so > 0.5) but not so high that adding the extra safe timeout (200 ms in this case) would exceed the second (so if you have XX.YYY and a safe timeout of 0.ZZZ  0.YYY + 0.ZZZ < 1.0)

Reply:
> An alternate approach would be to use `mtime` and `mtime.clock.os` libraries, which I've added to my mtime branch:
> 

`Mtime` makes sense.

> ```ocaml
>       match in_range ~e:Mtime.Span.(200 * ms) ~around:timeout elapsed with
>       | In_range ->
>           ()
> ...
> ```

No, this check is wrong, the interval you implemented is symmetric, if you set a timeout of 1 seconds for instance and the elapsed time is 0.999999 seconds it should be triggered as a bug.